daemon/networkdriver/bridge/driver.go

```diff
@@ -671,36 +658,24 @@ func AllocatePort(job *engine.Job) error {
 	}
 
 	if err != nil {
-		return err
+		return nat.PortBinding{}, err
 	}
 
 	network.PortMappings = append(network.PortMappings, host)
 
-	out := engine.Env{}
 	switch netAddr := host.(type) {
 	case *net.TCPAddr:
-		out.Set("HostIP", netAddr.IP.String())
-		out.SetInt("HostPort", netAddr.Port)
+		return nat.PortBinding{HostIp: netAddr.IP.String(), HostPort: strconv.Itoa(netAddr.Port)}, nil
 	case *net.UDPAddr:
-		out.Set("HostIP", netAddr.IP.String())
-		out.SetInt("HostPort", netAddr.Port)
-	}
-	if _, err := out.WriteTo(job.Stdout); err != nil {
-		return err
+		return nat.PortBinding{HostIp: netAddr.IP.String(), HostPort: strconv.Itoa(netAddr.Port)}, nil
+	default:
+		return nat.PortBinding{}, fmt.Errorf("unsupported address type %t", netAddr)
```

Comment:
Should be `%T` :) Yes, I'm checking with vet all PRs now until we merge autocheck.